### ISSUE TITLE
Establish a "connection string" concept for constructing Accumulo clients

### ISSUE THREAD

**Issue by ctubbsii** [apache/accumulo#748]
See https://github.com/apache/accumulo/issues/747#issuecomment-436349942

The idea here is to establish a connection string concept, so one can easily create an Accumulo client object from just a connection string, and can easily specify a connection string in a configuration field.

In the referenced comment, I proposed something like the following URI:

`accumulo://[user[:password]@]name?zkHosts=host1:port1,host2:port2,host3,host4:port4`

Other optional parameters could be added to the URI to support setting the path to a keytab file, or other authentication token information.

The `Accumulo.newClient()` entry point could have a method called `to(String connectionString)`, and we can document the connection string syntax fully in the Javadoc and on the website.

The client configuration file could have a `connectionString` property name. (maybe just `connection`?) Not sure the best name.

**Comment by mikewalch** [comment]
This could be implemented as an `instance.uri` property the `accumulo-client.properties` file:
```
## Instance properties
## -------------------

## Accumulo instance (accumulo://<instanceName>?zk=<zookeepers>)
## Other options:
##    zkt - ZooKeeper timeout (default: 30s)
instance.uri=accumulo://myinstance?zk=localhost:2181
```

**Comment by milleruntime** [comment]
I labeled this issue for 2.1.0 since I believe it is for future work and not going into 2.0.

**Comment by ctubbsii** [comment]
Might not go in 2.1.0 either. Maybe best to leave it unlabeled for now?

**Comment by milleruntime** [comment]
I was thinking it would be nice to know what tickets have been "triaged" aka addressed with labeling and having unlabeled issues was the only way I could think to do this.  

When we are looking to release 2.1 we could bump it to the next version, similar to how we did it in JIRA.

**Comment by ctubbsii** [comment]
In JIRA, we typically set a version if somebody wanted intended it to be fixed for that version, and then we bumped only if we couldn't get to it by the time we were ready to release. For many other issues, more "wish list" type items, or items that weren't well defined, or any specific timeline affiliated with them, we never set a version at all. For some issues that were bumped frequently, we removed the version entirely.

So, it varied a lot. Either way works for me... I just don't want to keep bumping (relabeling) low priority items that we never end up getting to.
